unixfs/src/file/visit.rs

```diff
@@ -0,0 +1,278 @@
+use cid::Cid;
+use std::borrow::Cow;
+use std::convert::TryFrom;
+use std::ops::Range;
+
+use crate::file::reader::{FileContent, FileReader, Traversal};
+use crate::file::{FileMetadata, FileReadFailed, UnwrapBorrowedExt};
+use crate::pb::merkledag::PBLink;
+
+/// IdleFileVisit represents a prepared file visit over a tree. The user has to know the CID and be
+/// able to get the block for the visit.
+#[derive(Default, Debug)]
+pub struct IdleFileVisit {
+    range: Option<Range<u64>>,
+}
+
+impl IdleFileVisit {
+    /// Target range represents the target byte range of the file we are interested in visiting.
+    pub fn with_target_range(self, range: Range<u64>) -> Self {
```

Comment:
this might be a personal preference, but when I see `x = x.method(y)` I expect it not to consume `self`, but to mutate it; I think `fn replace_range(&mut self, range: Range<u64>) { self.range = Some(range) }` or something more akin to that would be more clear

Reply:
I used `Self` return here to allow chaining to `IdleFileVisit::default().with_target_range(0..100).start(...)`.